bravenewcoin-vwap/adapter.js

```diff
@@ -0,0 +1,95 @@
+const { Requester, Validator } = require('@chainlink/external-adapter')
+
+const customParams = {
+  symbol: ['base', 'from', 'coin', 'symbol', 'assetId', 'indexId', 'asset'],
+  indexType: false,
+  timestamp: false
+}
+
+const host = 'bravenewcoin.p.rapidapi.com'
+
+const authenticate = async () => {
+  const response = await Requester.request({
+    method: 'POST',
+    url: `https://${host}/oauth/token`,
+    headers: {
+      'content-type': 'application/json',
+      'x-rapidapi-host': host,
+      'x-rapidapi-key': process.env.API_KEY,
+      accept: 'application/json',
+      useQueryString: true
+    },
```

Comment:
Let's reuse these repeated API headers. We can define an `apiHeaders` top-level constant:

```javascript
const apiHeaders = {
  'x-rapidapi-host': host,
  'x-rapidapi-key': process.env.API_KEY
}
```

And then add them here:

```suggestion
    headers: {
      'content-type': 'application/json',
      accept: 'application/json',
      useQueryString: true,
      ...apiHeaders
    },
```